products/firestore/api.yaml

```diff
@@ -0,0 +1,106 @@
+# Copyright 2019 Google Inc.
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+--- !ruby/object:Api::Product
+name: Firestore
+display_name: Cloud Firestore
+versions:
+  - !ruby/object:Api::Product::Version
+    name: ga
+    base_url: https://firestore.googleapis.com/v1/
+scopes:
+  - https://www.googleapis.com/auth/cloud-platform
+objects:
+  - !ruby/object:Api::Resource
+    name: 'Index'
+    base_url: projects/{{project}}/databases/{{database}}/collectionGroups/{{collection}}/indexes
+    self_link: '{{name}}'
+    input: true
+    description: |
+      Cloud Firestore indexes enable simple and complex queries against documents in a database.
+    references: !ruby/object:Api::Resource::ReferenceLinks
+      guides:
+        'Official Documentation': 'https://cloud.google.com/firestore/docs/query-data/indexing'
+      api: 'https://cloud.google.com/firestore/docs/reference/rest/v1/projects.databases.collectionGroups.indexes'
+    async: !ruby/object:Api::Async
+      operation: !ruby/object:Api::Async::Operation
+        path: 'name'
+        base_url: '{{op_id}}'
+        wait_ms: 1000
+        timeouts: !ruby/object:Api::Timeouts
+          insert_sec: 600
+          update_sec: 600
+          delete_sec: 600
+      result: !ruby/object:Api::Async::Result
+        path: 'response'
+        resource_inside_response: true
+      status: !ruby/object:Api::Async::Status
+        path: 'done'
+        complete: true
+        allowed:
+          - true
+          - false
+      error: !ruby/object:Api::Async::Error
+        path: 'error'
+        message: 'message'
+    properties:
+      - !ruby/object:Api::Type::String
+        name: name
+        output: true
+        description: |
+          A server defined name for this index. Format:
+          `projects/{{project}}/databases/{{database}}/collectionGroups/{{collection}}/indexes/{{server_generated_id}}`
+      - !ruby/object:Api::Type::String
+        name: database
+        default_value: '(default)'
+        description: |
+          The Firestore database id. Defaults to `"(default)"`.
+      - !ruby/object:Api::Type::String
+        name: collection
+        required: true
+        description: |
+          The collection being indexed.
+      - !ruby/object:Api::Type::Enum
+        name: queryScope
+        description: |
+          The scope at which a query is run. Defaults to `"COLLECTION"`.
+        default_value: :COLLECTION
+        values:
+          - :QUERY_SCOPE_UNSPECIFIED
```

Comment:
do we want to even let people set this? the docs say "Not a valid option."

Reply:
Ehh, I was leaning towards listing all the fields the API/docs do but I don't feel strongly. Removed.